deployments/kubernetes/install/scripts/install-cni.sh

```diff
@@ -20,25 +20,36 @@ function rm_bin_files() {
   echo "Removing existing binaries"
   rm -f /host/opt/cni/bin/istio-cni /host/opt/cni/bin/istio-iptables.sh
 }
-
-# The directory on the host where CNI networks are installed. Defaults to
-# /etc/cni/net.d, but can be overridden by setting CNI_NET_DIR.  This is used
-# for populating absolute paths in the CNI network config to assets
-# which are installed in the CNI network config directory.
-HOST_CNI_NET_DIR=${CNI_NET_DIR:-/etc/cni/net.d}
-MOUNTED_CNI_NET_DIR=${MOUNTED_CNI_NET_DIR:-/host/etc/cni/net.d}
-
-CNI_CONF_NAME_OVERRIDE=${CNI_CONF_NAME:-}
-
-# default to first file in `ls` output
-# if dir is empty, default to a filename that is not likely to be lexicographically first in the dir
-CNI_CONF_NAME=${CNI_CONF_NAME:-$(ls ${MOUNTED_CNI_NET_DIR} | head -n 1)}
-CNI_CONF_NAME=${CNI_CONF_NAME:-YYY-istio-cni.conflist}
-KUBECFG_FILE_NAME=${KUBECFG_FILE_NAME:-ZZZ-istio-cni-kubeconfig}
-CFGCHECK_INTERVAL=${CFGCHECK_INTERVAL:-1}
+# find_cni_conf_file
+#   Finds the CNI config file in the mounted CNI config dir.
+#   - Follows the same semantics as kubelet
+#     https://github.com/kubernetes/kubernetes/blob/954996e231074dc7429f7be1256a579bedd8344c/pkg/kubelet/dockershim/network/cni/cni.go#L144-L184
```

Comment:
kubelet covers a case where someone put a conflist into a .conf file.   Did you want to cover that case as well or leave for another change if needed.

Reply:
@john-a-joyce and I double-checked and kubelet isn't allowing putting a .conflist in a .conf file.  The code converts the .conf content to an internal .conflist value and uses it.